### PR TITLE
Feature : UI-improve

### PR DESCRIPTION
**Pull request by louiscrc**
This pull requests introduces a new color sets with a functional `dark mode`.

The colors are now modularized through the whole app, which allows us to modify easily the app thematic.
This is done by setting the color set in the `Theme.kt`, and calling `MaterialTheme.colorscheme` in any `composable` which uses colors.

There are also UI improvements like the `BottomNavigationBar`.

I also cleaned up the code from useless imports.
image image

### CODE REVIEW

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/Filter.kt [hunk 1]
```diff
@@ -700,11 +719,15 @@ fun FilterActionButton(
...
+                      contentColor = MaterialTheme.colorScheme.onPrimary,
                       disabledContainerColor =
-                          IcebreakrrBlue.copy(alpha = FILTER_ACTION_BUTTON_ALPHA))) {
-                Text("Filter", color = Color.White, fontSize = buttonTextSize)
+                          MaterialTheme.colorScheme.primary.copy(
+                              alpha = FILTER_ACTION_BUTTON_ALPHA))) {
+                Text(
+                    "Filter",
+                    color = MaterialTheme.colorScheme.onPrimary,
+                    fontSize = buttonTextSize)
```
[the remark this answers is not in the document]
Comment (louiscrc) [reply app/src/main/java/com/github/se/icebreakrr/ui/sections/Filter.kt:730]: But I fixed it though

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/Filter.kt [hunk 2]
```diff
@@ -327,26 +327,32 @@ fun FilterScreen(
...
                       GenderButton(
+                          label = "Other",
                           selected = otherSelected,
                           onClick = {
                             otherSelected = !otherSelected
                             isModified = checkModified()
                           },
-                          label = "Other")
+                          buttonWidth = buttonWidth,
+                          buttonHeight = buttonHeight)
```
[the remark this answers is not in the document]
Comment (louiscrc) [reply app/src/main/java/com/github/se/icebreakrr/ui/sections/Filter.kt:355]: Not from me again, but I fixed it

[the remark this answers is not in the document]
Comment (Kapestian) [reply app/src/main/java/com/github/se/icebreakrr/ui/sections/Filter.kt:355]: sorry, It is just that it appears in green on GitHub and my brain assumed it was new code even if I perfectly knew it was just changes of existing code. Don't worry about that

File: app/src/main/java/com/github/se/icebreakrr/ui/sections/Filter.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kapestian) [reply app/src/main/java/com/github/se/icebreakrr/ui/sections/Filter.kt:730]: sorryyyyyy

**Comment by louiscrc** [comment]
I implemented all your requests (even the ones not concerning my code) !

**Review by Kapestian — APPROVED** [review]
LGTM again

**Pull request merged**